content/docs/atom-guide/dev-guide/shaders/azsl/shader-variants-fallback-key.md

```diff
@@ -0,0 +1,134 @@
+---
+linkTitle: The Fallback Key
+title: Shader Variant Options and the Fallback Key
+description: Learn about the fallback key; the array that contains the shader variant options for an Amazon Shading Language (AZSL) shader. 
+weight: 100
+---
+
+Shader variant options are declared globally. They are encoded in a bit array as a member variable of *one, and only one* shader resource group (SRG). This bit array is known as the *fallback key*, and the SRG that owns the fallback key is known as the *shader variant fallback*.  
+  
+## Which SRG is chosen by the compiler to own the Fallback Key?
+The shader developer is the one who chooses which SRG will own the Fallback Key. This is done via the `ShaderVariantFallback` member of the SRG Semantic.  
+  
+If at least one Shader Variant Option is declared, then the developer must define the `ShaderVariantFallback` for one, and only one, SRG Semantic.  
+  
+In the following example we are declaring a single Shader Variant Option of boolean type, `o_useObjectColor`. There are two SRGs in this example. This example will fail to compile because the compilers needs one, and only one, SRG Semantic to define the `ShaderVariantFallback`.
+
+The compiler, **AZSLc**, will report the following error:  
+`IR error #129: If you have non-static options, one SRG must be designated as the default ShaderVariantFallback`.  
+```cpp
+    option bool o_useObjectColor;
+    
+    ShaderResourceGroupSemantic SRG_PerDraw
+    {
+        FrequencyId = 0;
+    }
+    ShaderResourceGroupSemantic SRG_PerObject
+    {
+        FrequencyId = 1;
+    }
+     
+    ShaderResourceGroup ObjectSrg : SRG_PerObject
+    {
+        float4 m_color;
+    }
+     
+    ShaderResourceGroup DrawSrg : SRG_PerDraw
+    {
+    	float4 m_color; 
+    }
+    
+    PSOutput MainPS(VSOutput IN)
+    {
+        PSOutput OUT;
+        if (o_useObjectColor)
+        {
+            OUT.m_color = ObjectSrg::m_color;
+        }
+        else
+        {
+            OUT.m_color = DrawSrg::m_color;
+        }
+        return OUT;
+    }
+```
+And here is the solution, in which the developer chooses `DrawSrg` to be the Shader Variant Fallback. In other words, `DrawSrg` will own the Fallback Key:  
```

Comment:
```suggestion
### Solution

In the following example, to resolve this, the developer chooses `DrawSrg` as the shader variant fallback. In other words, `DrawSrg` owns the fallback key:  

```

* Lower case: "shader variant fallback" and "fallback key"

Reply:
Fixed.